docs/interactive_engine/tutorial_ldbc_gremlin.md

```diff
@@ -831,3 +831,1087 @@ At last, we would like to introduce `endV()` step in path expansion. Since a pat
+                     .where(\'university1\', eq(\'university2\'))\
+                     .select(\'person1\', \'person2\', \'university1\')')
+print(q1.all())
+```
+
+- At first, you find all pairs of `person1` and `person2` they know each other
+- Then you find `person1's` university as `university1`, and `person2`'s university as `university2`.
+- Then you use `where(...)` step to check whether `university1` is equal to `university2`, and filter out those unequal ones.
+- Finally, you use `select(...)` step to project `person1`, `person2` and `university1` (now `university1` is the same as `university1`)
+
+The output of should be like:
+
+```bash
+# Person1 and 2 know each other and study at the same university
+[{'person2': v[216172782113784481],
+  'person1': v[216172782113784279],
+  'university1': v[144115188075858884]},
+ {'person2': v[216172782113784361],
+  'person1': v[216172782113784291],
+  'university1': v[144115188075858879]},
+ {'person2': v[216172782113784642],
+  'person1': v[216172782113784291],
+  'university1': v[144115188075858879]},
+ {'person2': v[216172782113784473],
+  'person1': v[216172782113784328],
+  'university1': v[144115188075858700]},
+ {'person2': v[216172782113784700],
+  'person1': v[216172782113784331],
+  'university1': v[144115188075860619]},
+ {'person2': v[216172782113784375],
+  'person1': v[216172782113784333],
+  'university1': v[144115188075858813]},
+ {'person2': v[216172782113784593],
+  'person1': v[216172782113784361],
+  'university1': v[144115188075858879]},
+ {'person2': v[216172782113784642],
+  'person1': v[216172782113784361],
+  'university1': v[144115188075858879]},
+ {'person2': v[216172782113784291],
+...
+  'person1': v[216172782113784688],
+  'university1': v[144115188075860870]},
+ {'person2': v[216172782113784047],
+  'person1': v[216172782113784692],
+  'university1': v[144115188075862429]}]
+```
+
+Actually, such kind of problem is called pattern match. It means, you define a pattern at first. The pattern can be either described by the natural language:
+
+- "Find two `persons` and the `universities` they `studyAt` in the Property Graph that the two `person` `know` each other and the two `person` `studyAt` the same `university`"
+
+or can be illustrated as a figure:
+
+:::{figure-md}
+<img src="../images/pattern_person_person_university.png" alt="pattern_person_person_university" style="zoom: 50%;" />
+Figure 21. Pattern: two person know each other and study at the same university.
+:::
+
+Then you aim to find all the matched subgraphs of the Property Graph that are isomorphic to the pattern. For example, this is a sub-local-graph of university vertex(id = 144115188075858884) in the LDBC Graph: 
+
+:::{figure-md}
+<img src="../images/local_person_person_university.png" alt="local_person_person_university" style="zoom:50%;" />
+Figure 22. sub-local-graph of a university vertex. 
+:::
+
+and here are the matched subgraphs in this sub-local-graph:
```

Comment:
`here are the matched subgraphs in this sub-local-graph`. In the following, it occurs twice. Rephrase these sentences a bit.